packages/web-components/src/components/tabs-extended/tabs-extended.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+} from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import styles from './tabs-extended.scss';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+import DDSTab from './tab';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * A component to present content inside a tabbed layout.
+ *
+ * @element dds-tabs-extended
+ */
+@customElement(`${ddsPrefix}-tabs-extended`)
+class DDSTabsExtended extends StableSelectorMixin(LitElement) {
+  /**
+   * Child tab components.
+   */
+  @internalProperty()
+  private _tabItems: Node[] = [];
+
+  /**
+   * Defines the active tab index.
+   */
+  @internalProperty()
+  private active = 0;
+
+  /**
+   * Handler for @slotChange, creates tabs from dds-tab components.
+   *
+   * @private
+   */
+  private _handleSlotChange(event: Event) {
+    this._tabItems = (event.target as HTMLSlotElement)
+      .assignedNodes({ flatten: true })
+      .filter(node => (new DDSTab)?.nodeName === node.nodeName);
+    this._tabItems.forEach((tab:DDSTab, index) => {
+      this.active = tab.selected ? index : this.active;
+    });
+  }
+
+  private _setActiveTab(index) {
+    this.active = index;
+  }
+
+  updated() {
+    this._tabItems.map((tab:DDSTab, index) => {
+      tab.selected = (index === this.active);
+      tab.setIndex(index);
+    });
+  }
+
+  render() {
+    const {
+      _tabItems: tabs,
+    } = this;
+    return html`
+      <div class="${prefix}--tabs-extended">
+
+        <div class="${prefix}--accordion">
+          <ul data-accordion class="bx--accordion">
+            ${tabs.map((tab:DDSTab, index) => {
+              const classes = [
+                index === this.active && `${prefix}--accordion__item--active` || null,
+              ];
```

Comment:
Should use: https://lit-html.polymer-project.org/guide/template-reference#classmap